lib/character.rb

```diff
@@ -0,0 +1,26 @@
+class Character
+
+  attr_reader :hash
+
+  def initialize(hash)
+    # @name = hash[:name]
+    # @actor = hash[:actor]
+    # @salary = hash[:salary]
+    @hash = hash
+    # @characters_array = [Character.new("KITT", "William Daniels", 1000000), Character.new({name: "Michael Knight", actor: "David Hasselhoff", salary: 1_600_000})]
+
+  end
+
+  def name
+    @hash[:name]
+  end
+
+  def actor
+    @hash[:actor]
+  end
+
+  def salary
+    @hash[:salary]
+  end
+
```

Comment:
these methods work okay, but it'd be less work to use the `@name = hash[:name]` etc code that you have commented out in the initialize method, and make attr_readers of those instance variables. Also, the hash of data itself doesn't need to be an attr_reader since we never require that that data gets exposed from our class.